src/sentinel1_reader/sentinel1_reader.py

```diff
@@ -0,0 +1,418 @@
+        sensing_starts[i] = sensing_start
+        sensing_times[i] = as_datetime(burst_list_element.find('sensingTime').text)
+        dt = sensing_times[i] - ascending_node_time
+        id_burst = int((dt.seconds + dt.microseconds / 1e6) // burst_interval)
+
+        # choose nearest azimuth FM rate
+        d_seconds = 0.5 * (n_lines - 1) * azimuth_time_interval
+        sensing_mid = sensing_start + datetime.timedelta(seconds=d_seconds)
+        az_fm_rate = get_nearest_polynomial(sensing_mid, az_fm_rate_list)
+
+        # choose nearest doppler
+        poly1d = get_nearest_polynomial(sensing_mid, doppler_list)
+        lut2d = doppler_poly1d_to_lut2d(poly1d, starting_range,
+                                        range_pxl_spacing, (n_lines, n_samples),
+                                        azimuth_time_interval)
+        lut1d = isce3.core.avg_lut2d_to_lut1d(lut2d)
+        doppler = Doppler(poly1d, lut2d)
+
+        # get orbit from state vector list/element tree
+        sensing_duration = datetime.timedelta(
+            seconds=n_samples * azimuth_time_interval)
+        orbit = get_burst_orbit(sensing_start, sensing_start + sensing_duration,
+                                osv_list)
+
+        # determine burst offset and dimensions
+        # TODO move to own method
+        first_valid_samples = [int(val) for val in burst_list_element.find('firstValidSample').text.split()]
+        last_valid_samples = [int(val) for val in burst_list_element.find('lastValidSample').text.split()]
+
+        first_valid_line = [x > 0 for x in first_valid_samples].index(True)
+        n_valid_lines = [x > 0 for x in first_valid_samples].count(True)
+        last_line = first_valid_line + n_valid_lines - 1
+
+        first_valid_sample = max(first_valid_samples[first_valid_line],
+                                 first_valid_samples[last_line])
+        last_sample = min(last_valid_samples[first_valid_line],
+                          last_valid_samples[last_line])
+        n_valid_samples = last_sample - first_valid_sample
+
+        burst_id = f't{track_number}_{subswath_id.lower()}_b{id_burst}'
+
+        bursts[i] = Sentinel1BurstSlc(sensing_start, radar_freq, wavelength,
+                                      azimuth_steer_rate, azimuth_time_interval,
+                                      slant_range_time, starting_range,
+                                      range_sampling_rate, range_pxl_spacing,
+                                      (n_lines, n_samples), az_fm_rate, doppler,
+                                      rng_processing_bandwidth, pol, burst_id,
+                                      platform_id, center_pts[i], boundary_pts[i],
+                                      orbit, tiff_path, i, first_valid_sample,
+                                      last_sample, first_valid_line, last_line)
+
+    return bursts
+
+def zip2bursts(zip_path: str, osv_list: ET.Element,
```

Comment:
Sorry for nitpicking. Could we have a better name? How about "read_bursts" ? @vbrancat @yunjunz do you have other suggestions?